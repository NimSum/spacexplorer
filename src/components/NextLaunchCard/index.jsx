import React, { Component } from 'react'

const NextLaunchCard = ({ launch }) => {
  return (
    <article className='next-launch-card'>
 
    </article>
  )
}

export default NextLaunchCard

// id(pin): "73a3c33c-5679-4f64-aa03-67167cb0a7e1"
// url(pin): "https://spacelaunchnow.me/api/3.3.1/launch/73a3c33c-5679-4f64-aa03-67167cb0a7e1/"
// launch_library_id(pin): 1649
// slug(pin): "https://spacelaunchnow.me/launch/proton-mbriz-m-yamal-601-1649"
// name(pin): "Proton-M/Briz-M | Yamal-601"
// img_url(pin): null
// net(pin): "2019-05-30T17:42:00Z"
// window_end(pin): "2019-05-30T17:42:00Z"
// window_start(pin): "2019-05-30T17:42:00Z"
// inhold(pin): false
// tbdtime(pin): false
// tbddate(pin): false
// probability(pin): -1
// holdreason(pin): null
// failreason(pin): null
// hashtag(pin): null