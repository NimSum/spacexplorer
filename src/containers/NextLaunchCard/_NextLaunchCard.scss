article.next-launch-card {
  min-width: 310px;
  white-space: normal;
  margin: 0 10px;
  display: flex;
  background-color: $secondary-light;
  border-radius: 5px;
  padding: 10px;

  h3 {
    margin: 5px 0;
    padding: 0 5px;
    color: $primary-dark;
  }

  p {
    margin: 5px 0;
  }

  div.image-side {
    flex: 1.5;
    text-align: center;
    display: flex;
    justify-content: space-around;
    flex-direction: column;
    
    img {
      width: 100%;
      height: 80%;
      object-fit: cover;
      border-radius: 5px;
    }
  }

  div.info-side {
    flex: 2;
    position: relative;
    
    p {
      margin: 10px;
    }

    h3 {
      margin: 14px 0;
    }

    button {
      position: absolute;
      bottom: 10px;
      right: 20px;
      @include more-info-btn;
    }
  }
}