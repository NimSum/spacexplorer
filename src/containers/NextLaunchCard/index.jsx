import React from 'react';
import { connect } from 'react-redux';
import { toggleLaunchInfo, addSelectedLaunch } from '../../actions';

const NextLaunchCard = ({ launch, toggleLaunchInfo, updateSelectedLaunch }) => {
  const { name, pad, status, mission } = launch;
  const showSelectedLaunch = () => {
    toggleLaunchInfo(true);
    updateSelectedLaunch(launch);
  }
  return (
    <article className='next-launch-card'>
      <h3>{ name }</h3>
      <p>Orbit: { mission ? mission.orbit : 'Uknown' }</p>
      <p>Status: { status.name }</p>
      <p>Location: { pad.location.name }</p>
      <button onClick={ showSelectedLaunch }>
        More
      </button>
    </article>
  )
}

export const mapDispatchToProps = dispatch => ({
  toggleLaunchInfo: (bool) => dispatch(toggleLaunchInfo(bool)),
  updateSelectedLaunch: (launch) => dispatch(addSelectedLaunch(launch))
})

export default connect(null, mapDispatchToProps)(NextLaunchCard);
