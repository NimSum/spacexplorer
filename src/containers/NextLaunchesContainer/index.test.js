import React from 'react';
import { NextLaunchesContainer, mapStateToProps } from './index';
import { shallow } from 'enzyme';
import { mockLaunches } from '../../utils/mockData';
import NextLaunchCard from '../NextLaunchCard';

describe('NextLaunchesContainer', () => {
  const defaultState = { showInfo: false }
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      < NextLaunchesContainer
        rocketLaunches={ mockLaunches }
      />
    )
  })

  it('should match component snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have default state', () => {
    expect(wrapper.state()).toEqual(defaultState);
  })

  it('should generate launch cards', () => {
    const modifiedData = { ...mockLaunches };
    modifiedData.results = [mockLaunches.results[0]];
    const expected = [
      < NextLaunchCard 
        launch={modifiedData.results[0] } 
        key={ modifiedData.results[0].id }/>
    ];
    wrapper.setProps({ rocketLaunches: modifiedData });
    const result = wrapper.instance().generateCards();
    expect(result).toEqual(expected);
  })

  it('should be able to toggle showInfo state', () => {
    wrapper.instance().toggleInfo();
    expect(wrapper.state().showInfo).toBe(true);
    wrapper.instance().toggleInfo();
    expect(wrapper.state().showInfo).toBe(false);
  })
})