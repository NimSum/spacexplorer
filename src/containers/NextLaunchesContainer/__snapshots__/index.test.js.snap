// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextLaunchesContainer should match component snapshot 1`] = `
<section
  className="next-launches-container"
>
  <Connect(NextLaunch) />
  <h3
    className="next-launches-title"
  >
    Upcoming Launches
  </h3>
  <div
    className="card-container"
  >
    <Connect(NextLaunchCard)
      key="41077f29-434a-4e53-9360-ca754f37e052"
      launch={
        Object {
          "failreason": null,
          "hashtag": null,
          "holdreason": null,
          "id": "41077f29-434a-4e53-9360-ca754f37e052",
          "img_url": null,
          "infoURLs": Array [],
          "inhold": false,
          "launch_library_id": 1927,
          "mission": Object {
            "description": "The first sea launch of Long March 11 rocket. It carries two Jilin-1 Earth observation satellites.",
            "id": 990,
            "launch_library_id": 1200,
            "name": "Jilin-1",
            "orbit": "Sun-Synchronous Orbit",
            "orbit_abbrev": "SSO",
            "type": "Earth Science",
          },
          "name": "Long March 11 | Jilin-1",
          "net": "2019-06-05T04:00:00Z",
          "pad": Object {
            "agency_id": null,
            "id": 101,
            "info_url": null,
            "latitude": "0",
            "location": Object {
              "country_code": "UNK",
              "id": 3,
              "name": "Sea Launch",
            },
            "longitude": "0",
            "map_url": "",
            "name": "Unknown Pad",
            "wiki_url": "",
          },
          "probability": -1,
          "rocket": Object {
            "configuration": Object {
              "alias": "",
              "apogee": 200,
              "description": "",
              "diameter": 2,
              "family": "Long March",
              "full_name": "Long March 11",
              "gto_capacity": 0,
              "id": 47,
              "image_url": "https://daszojo4xmsc6.cloudfront.net/media/launcher_images/long2520march252011_image_20190222031217.jpeg",
              "info_url": null,
              "launch_library_id": 71,
              "launch_mass": 58,
              "launch_service_provider": Object {
                "abbrev": "CNSA",
                "administrator": "Administrator: Tang Dengjie",
                "country_code": "CHN",
                "description": "The China National Space Administration (CNSA) is the national space agency of the People's Republic of China. It is responsible for the national space program and for planning and development of space activities. CNSA and China Aerospace Corporation (CASC) assumed the authority over space development efforts previously held by the Ministry of Aerospace Industry. The CNSA has many launch sites around China with their orbital sites located in Jiuquan, Xichang, Taiyuan and Wenchang.",
                "failed_launches": 1,
                "featured": true,
                "founding_year": "1993",
                "id": 17,
                "image_url": "https://daszojo4xmsc6.cloudfront.net/media/agency_images/china2520national2520space2520administration_image_20190207032431.jpeg",
                "info_url": "http://www.cnsa.gov.cn/",
                "launch_library_url": "https://launchlibrary.net/1.4/agency/17",
                "launchers": "Long March",
                "logo_url": "https://daszojo4xmsc6.cloudfront.net/media/logo/china2520national2520space2520administration_logo_20190207032431.png",
                "name": "China National Space Administration",
                "nation_url": "https://daszojo4xmsc6.cloudfront.net/media/agency_nation/china2520national2520space2520administration_nation_20190602114400.png",
                "pending_launches": 2,
                "spacecraft": "Shenzhou",
                "successful_launches": 62,
                "type": "Government",
                "url": "https://spacelaunchnow.me/api/3.3.1/agencies/17/",
                "wiki_url": "http://en.wikipedia.org/wiki/China_National_Space_Administration",
              },
              "length": 20,
              "leo_capacity": 500,
              "maiden_flight": null,
              "max_stage": 4,
              "min_stage": 1,
              "name": "Long March 11",
              "to_thrust": 1177,
              "url": "https://spacelaunchnow.me/api/3.3.1/config/launcher/47/",
              "variant": "11",
              "vehicle_range": 200,
              "wiki_url": "https://en.wikipedia.org/wiki/Long_March_11",
            },
            "launcher_stage": Array [],
            "spacecraft_stage": null,
          },
          "slug": "https://spacelaunchnow.me/launch/long-march-11-jilin-1-1927",
          "status": Object {
            "id": 1,
            "name": "Go",
          },
          "tbddate": false,
          "tbdtime": false,
          "url": "https://spacelaunchnow.me/api/3.3.1/launch/41077f29-434a-4e53-9360-ca754f37e052/",
          "vidURLs": Array [],
          "window_end": "2019-06-05T04:00:00Z",
          "window_start": "2019-06-05T04:00:00Z",
        }
      }
    />
    <Connect(NextLaunchCard)
      key="7bc965d8-9821-472e-aeba-f521ae2c06cf"
      launch={
        Object {
          "failreason": null,
          "hashtag": null,
          "holdreason": null,
          "id": "7bc965d8-9821-472e-aeba-f521ae2c06cf",
          "img_url": null,
          "infoURLs": Array [],
          "inhold": false,
          "launch_library_id": 1931,
          "mission": null,
          "name": "Hyperbola-1 | 7 unknown payloads",
          "net": "2019-06-10T02:00:00Z",
          "pad": Object {
            "agency_id": null,
            "id": 71,
            "info_url": null,
            "latitude": "40.958",
            "location": Object {
              "country_code": "CHN",
              "id": 17,
              "name": "Jiuquan, People's Republic of China",
            },
            "longitude": "100.291",
            "map_url": "",
            "name": "Unknown Pad, Jiuquan",
            "wiki_url": "",
          },
          "probability": -1,
          "rocket": Object {
            "configuration": Object {
              "alias": "",
              "apogee": null,
              "description": "i-Space's first orbital rocket.",
              "diameter": null,
              "family": "Hyperbola",
              "full_name": "Hyperbola-1",
              "gto_capacity": null,
              "id": 193,
              "image_url": null,
              "info_url": null,
              "launch_library_id": 222,
              "launch_mass": null,
              "launch_service_provider": Object {
                "abbrev": "iSpace",
                "administrator": null,
                "country_code": "CHN",
                "description": "i-Space is a Chinese private space launch company based in Beijing and founded in October 2016.",
                "failed_launches": 0,
                "featured": false,
                "founding_year": "2016",
                "id": 274,
                "image_url": null,
                "info_url": null,
                "launch_library_url": "https://launchlibrary.net/1.4/agency/274",
                "launchers": "Hyperbola-1",
                "logo_url": null,
                "name": "iSpace",
                "nation_url": "https://daszojo4xmsc6.cloudfront.net/media/agency_nation/ispace_nation_20190602114618.png",
                "pending_launches": 1,
                "spacecraft": "",
                "successful_launches": 0,
                "type": null,
                "url": "https://spacelaunchnow.me/api/3.3.1/agencies/274/",
                "wiki_url": "https://en.wikipedia.org/wiki/I-Space_(Chinese_company)",
              },
              "length": null,
              "leo_capacity": 300,
              "maiden_flight": null,
              "max_stage": null,
              "min_stage": null,
              "name": "Hyperbola-1",
              "to_thrust": null,
              "url": "https://spacelaunchnow.me/api/3.3.1/config/launcher/193/",
              "variant": "1",
              "vehicle_range": null,
              "wiki_url": null,
            },
            "launcher_stage": Array [],
            "spacecraft_stage": null,
          },
          "slug": "https://spacelaunchnow.me/launch/hyperbola-1-7-unknown-payloads-1931",
          "status": Object {
            "id": 2,
            "name": "TBD",
          },
          "tbddate": true,
          "tbdtime": true,
          "url": "https://spacelaunchnow.me/api/3.3.1/launch/7bc965d8-9821-472e-aeba-f521ae2c06cf/",
          "vidURLs": Array [],
          "window_end": "2019-06-10T02:00:00Z",
          "window_start": "2019-06-10T02:00:00Z",
        }
      }
    />
    <Connect(NextLaunchCard)
      key="6619844e-4777-4f7f-81d6-9ef0c3308d1d"
      launch={
        Object {
          "failreason": null,
          "hashtag": null,
          "holdreason": null,
          "id": "6619844e-4777-4f7f-81d6-9ef0c3308d1d",
          "img_url": null,
          "infoURLs": Array [],
          "inhold": false,
          "launch_library_id": 1937,
          "mission": Object {
            "description": "First flight of the new solid launcher developed by ExPace, subsidiary of CASIC. It will carry 2 communication satellites on this launch.",
            "id": 1003,
            "launch_library_id": 1212,
            "name": "Maiden Flight",
            "orbit": "Low Earth Orbit",
            "orbit_abbrev": "LEO",
            "type": "Test Flight",
          },
          "name": "Kuaizhou-11 | Maiden Flight",
          "net": "2019-06-15T00:00:00Z",
          "pad": Object {
            "agency_id": null,
            "id": 71,
            "info_url": null,
            "latitude": "40.958",
            "location": Object {
              "country_code": "CHN",
              "id": 17,
              "name": "Jiuquan, People's Republic of China",
            },
            "longitude": "100.291",
            "map_url": "",
            "name": "Unknown Pad, Jiuquan",
            "wiki_url": "",
          },
          "probability": -1,
          "rocket": Object {
            "configuration": Object {
              "alias": "",
              "apogee": null,
              "description": "",
              "diameter": null,
              "family": "",
              "full_name": "",
              "gto_capacity": null,
              "id": 194,
              "image_url": null,
              "info_url": "",
              "launch_library_id": 223,
              "launch_mass": null,
              "launch_service_provider": Object {
                "abbrev": "EP",
                "administrator": null,
                "country_code": "CHN",
                "description": null,
                "failed_launches": 0,
                "featured": false,
                "founding_year": null,
                "id": 194,
                "image_url": null,
                "info_url": "",
                "launch_library_url": "https://launchlibrary.net/1.4/agency/194",
                "launchers": "",
                "logo_url": null,
                "name": "ExPace",
                "nation_url": null,
                "pending_launches": 1,
                "spacecraft": "",
                "successful_launches": 0,
                "type": "Commercial",
                "url": "https://spacelaunchnow.me/api/3.3.1/agencies/194/",
                "wiki_url": "https://en.wikipedia.org/wiki/ExPace",
              },
              "length": null,
              "leo_capacity": null,
              "maiden_flight": null,
              "max_stage": null,
              "min_stage": null,
              "name": "Kuaizhou-11",
              "to_thrust": null,
              "url": "https://spacelaunchnow.me/api/3.3.1/config/launcher/194/",
              "variant": "",
              "vehicle_range": null,
              "wiki_url": "",
            },
            "launcher_stage": Array [],
            "spacecraft_stage": null,
          },
          "slug": "https://spacelaunchnow.me/launch/kuaizhou-11-maiden-flight-1937",
          "status": Object {
            "id": 2,
            "name": "TBD",
          },
          "tbddate": true,
          "tbdtime": true,
          "url": "https://spacelaunchnow.me/api/3.3.1/launch/6619844e-4777-4f7f-81d6-9ef0c3308d1d/",
          "vidURLs": Array [],
          "window_end": "2019-06-15T00:00:00Z",
          "window_start": "2019-06-15T00:00:00Z",
        }
      }
    />
  </div>
</section>
`;

exports[`NextLaunchesContainer should match snapshot test with no rocket launches to render 1`] = `
<section
  className="next-launches-container"
>
  <Connect(NextLaunch) />
  <h3
    className="next-launches-title"
  >
    Upcoming Launches
  </h3>
  <div
    className="card-container"
  />
</section>
`;
